relay-spans/src/v2_to_v1.rs

```diff
@@ -270,6 +277,178 @@ fn derive_op_for_v2_span(span: &SpanV2) -> String {
+        Some(SpanV2Kind::Client) => get_client_url_path(attributes),
+        _ => None,
+    };
+
+    if url_path.is_none() {
+        return Some(description);
+    }
+
+    let url_path = url_path.unwrap();
+    let base_description = format!("{} {}", http_method, url_path);
+
+    // Check for GraphQL operations
+    if let Some(graphql_ops) = attributes
+        .get("sentry.graphql.operation")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(format!("{} ({})", base_description, graphql_ops));
+    }
+
+    Some(base_description)
+}
+
+fn derive_db_description(attributes: &Object<Attribute>) -> Option<String> {
+    // `db.query.text` is a recommended attribute, and it contains the full query text if available.
+    // This is the ideal description.
+    if let Some(query_text) = attributes
+        .get("db.query.text")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(query_text.to_string());
+    }
+
+    // Other SDKs check for `db.statement`, it's a legacy OTel attribute, useful as a fallback in some cases.
+    if let Some(statement) = attributes
+        .get("db.statement")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(statement.to_string());
+    }
+
+    None
+}
+
+fn get_server_url_path(attributes: &Object<Attribute>) -> Option<String> {
+    // `http.route` takes precedence. If available, this is the matched route of the server
+    // framework for server spans. Not always available, even for server spans.
+    if let Some(route) = attributes
+        .get("http.route")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(route.to_string());
+    }
+
+    // `url.path` is the path of the HTTP request for server spans. This is required for server spans.
+    if let Some(path) = attributes
+        .get("url.path")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(path.to_string());
+    }
+
+    // `http.target` is deprecated, but might be present in older data. Here as a fallback
+    if let Some(target) = attributes
+        .get("http.target")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(strip_url_query_and_fragment(target));
+    }
+
+    None
+}
+
+fn strip_url_query_and_fragment(url: &str) -> String {
+    url.split('?')
+        .next()
+        .unwrap_or(url)
+        .split('#')
+        .next()
+        .unwrap_or(url)
+        .to_string()
```

Comment:
`split` lets you pass multiple characters at once:
```suggestion
    url.split(&['?', '#']).next().unwrap_or(url).to_owned()
```